pkg/k8scontext/context.go

```diff
@@ -350,7 +354,64 @@ func (c *Context) GetVirtualServicesForGateway(gateway v1alpha3.Gateway) []*v1al
 	return virtualServices
 }
 
-func isIngressApplicationGateway(ingress *v1beta1.Ingress) bool {
+// SetIngressStatuses sets IP address in Ingress Status. This is best effort
+func (c *Context) SetIngressStatuses(addIngressStatusIngressList []*v1beta1.Ingress, removeIngressStatusIngressList []*v1beta1.Ingress, address string) {
+	for _, ingress := range addIngressStatusIngressList {
+		// best effort: start independent go routines to update Ingress address
+		go c.AddIngressStatus(ingress, address)
```

Comment:
This is very interesting - I love the async approach.
One concern is the passing of a pointer to a goroutine. I suspect that through the life of the goroutine we could lose allocation.
Here is an example - https://play.golang.org/p/r2rwAPnysZt